packages/core/src/Modules/Cashier/Containers/on-ramp.jsx

```diff
@@ -1,56 +1,143 @@
+                                )}
+                                collapse_length={140}
+                                className='on-ramp__read-more'
+                                text_props={{ size: 'xxs', line_height: 'm' }}
+                            />
+                        </div>
+                    </div>
+                    <Text
+                        weight='bold'
+                        color='less-prominent'
+                        align='center'
+                        line_height='m'
+                        className='on-ramp__page-header'
+                        as='p'
+                    >
+                        <Localize i18n_default_text='Select payment channel' />
+                    </Text>
+                    {filtered_onramp_providers.map(provider => (
+                        <OnRampProviderCard key={provider.name} provider={provider} />
+                    ))}
+                    <Modal
+                        className='on-ramp__modal'
+                        has_close_icon
+                        is_open={is_onramp_modal_open}
+                        small={should_show_dialog}
+                        title={onramp_popup_modal_title}
+                        toggleModal={() => setIsOnRampModalOpen(!is_onramp_modal_open)}
+                        onUnmount={resetPopup}
+                        width={should_show_dialog ? '440px' : '628px'}
+                    >
+                        <Modal.Body>
+                            <OnRampProviderPopup />
+                        </Modal.Body>
+                    </Modal>
+                </div>
+            </MobileWrapper>
+            <DesktopWrapper>
+                <div className='cashier__wrapper cashier__wrapper--align-left on-ramp'>
+                    <Text weight='bold' color='general' line_height='m' className='on-ramp__page-header' a='p'>
+                        <Localize i18n_default_text='Select payment channel' />
+                    </Text>
+                    {filtered_onramp_providers.map(provider => (
+                        <OnRampProviderCard key={provider.name} provider={provider} />
+                    ))}
+                    <Modal
+                        className='on-ramp__modal'
+                        has_close_icon
+                        is_open={is_onramp_modal_open}
+                        small={should_show_dialog}
+                        title={onramp_popup_modal_title}
+                        toggleModal={() => setIsOnRampModalOpen(!is_onramp_modal_open)}
+                        onUnmount={resetPopup}
+                        width={should_show_dialog ? '440px' : '628px'}
+                    >
+                        <Modal.Body>
+                            <OnRampProviderPopup />
+                        </Modal.Body>
```

Comment:
Since this is pretty much the same as for mobile you should consider creating another component.